nandtool/nandtool.c

```diff
@@ -177,6 +177,91 @@ static int nandtool_check(void)
 }
 
 
+static int nandtool_dump(const char *outpath, unsigned int start, unsigned int nblocks, int oob)
+{
+	flashsrv_info_t *info;
+	int ret;
+	size_t blocksz, partsz, pagesz;
+	size_t addr, endaddr;
```

Comment:
I think we already discussed it - but the addresses should be of type `off_t` to be able to get to data > 2GB. If I remember correctly - change in the driver interface is needed, so I'm ok with writing FIXME/TODO for now in places the change would be needed

Reply:
Yes, the problem lies in devctl msg API to `flashsrv`, which accepts a 32-bit integer. I'll change it here to `off_t` and change the parameter type of `flashmng_readraw`, so that it accepts the `off_t` type. It won't be consistent with other `flashmng_` functions (that's why I left it as it is), but for sure it would be a small step forward :)